langchain-core/src/_standard/message.ts

```diff
@@ -104,3 +104,21 @@ export type SystemMessage<
 export type ToolMessage<
   TComplex extends $MessageComplex = $StandardMessageComplex
 > = BaseMessage<TComplex, "tool">;
+
+export function isAIMessage(message: BaseMessage): message is AIMessage {
+  return message.type === "ai";
+}
+
+export function isHumanMessage(message: BaseMessage): message is HumanMessage {
+  return message.type === "user";
+}
+
+export function isSystemMessage(
+  message: BaseMessage
+): message is SystemMessage {
+  return message.type === "system";
+}
+
+export function isToolMessage(message: BaseMessage): message is ToolMessage {
+  return message.type === "tool";
+}
```

Comment:
think it would be better to type the message params as `message: unknown` and make some more assertions that the thing were testing is _actually_ a message

```suggestion
export function isBaseMessage(message: unknown): message is BaseMessage {
  ...
}

export function isAIMessage(message: unknown): message is AIMessage {
  return isBaseMessage(message) && message.type === "ai";
}

export function isHumanMessage(message: unknown): message is HumanMessage {
  return isBaseMessage(message) && message.type === "user";
}

export function isSystemMessage(
  message: unknown
): message is SystemMessage {
  return isBaseMessage(message) && message.type === "system";
}

export function isToolMessage(message: BaseMessage): message is ToolMessage {
  return message.type === "tool";
}
```